samples/bluetooth/iso_receive/prj.conf

```diff
@@ -5,3 +5,4 @@ CONFIG_LOG=y
 CONFIG_BT_DEVICE_NAME="Test ISO Receive"
 
 CONFIG_BT_ISO_MAX_CHAN=2
+CONFIG_GPIO=y
```

Comment:
This PR is doing something similar to what https://github.com/zephyrproject-rtos/zephyr/pull/81435 attempted, and my review comment is the same:
The GPIO usage in many/most of these samples are optional, i.e. it most cases I am aware of, it checks if there is an LED and if there is, it toggles that. 

Enabling GPIO in the prj.conf file will provide build warnings/errors for boards that does not have that, even though the boards can run the sample without the LED toggle just fine. 

In other words, GPIO is optional in many of these samples, and this is effectively making it mandatory. 

I'm not familiar with all samples, and some may actually require it, but the `iso_receive` and `periodic_sync` samples only use LED if it is there, and if there's no LED there's no GPIO usage. 

@cvinayak I think you've added this for some of these samples. What are your thoughts?

Reply:
For such cases, the sample should:

- Have a Kconfig option that selects GPIO if LED exists and enabled in DT
- Or, have an alternate config (e.g. in prj_gpio.conf) that can be used to test with GPIO enabled

Which one is preferred?

Reply:
> > if a board has a LED defined, build will fail if it doesn't have CONFIG_GPIO=y, so this is wrong.
> 
> Is it possible that `DT_NODE_HAS_STATUS_OKAY(LED0_NODE)` returns `1` without GPIO? :o
> 

Yes, this is just checking DT. A LED node could implement other APIs, in fact, e.g. LED API.

> In which case, the check should be `DT_NODE_HAS_STATUS_OKAY(LED0_NODE) && IS_ENABLED(CONFIG_GPIO)`.

That works for me. I'll update the PR.

Reply:
> That works for me. I'll update the PR.

Great - But we will need to determine where/how to enable GPIO :)

Reply:
please check latest patch set

Reply:
Now it's back to only being enable by the board itself, isn't it? So once GPIO is removed from boards, it's not longer automatically enabled for board that support the feature? This seems contrary to what the purpose of this PR is supposed to do

Reply:
I've added a sample variant that enables GPIO, isn't that sufficient? If not you need to create an auxiliary Kconfig option that defaults to y if DT has a LED node enabled, selects GPIO and is used as a condition to compile that GPIO branches.

Reply:
@Thalley check latest patch set, I think it is better.

Reply:
> @Thalley check latest patch set, I think it is better.

Yeah, that looks good to me

Should that be applied to all samples where you've set `CONFIG_GPIO=y`?

Reply:
Some samples use GPIO unconditionally, so not always. It's true that we could probably find some common patterns and simplify some samples.